socorro/external/postgresql/setupdb_app.py

```diff
@@ -410,6 +346,85 @@ def create_table(element, compiler, **kw):
+            return 1
+
+        # superuser credentials for overall database
+        superuser_pg_url = self.create_connection_url(
+            'postgres',
+            self.config.database_superusername,
+            self.config.database_superuserpassword
+        )
+
+        # superuser credentials for working database
+        superuser_normaldb_pg_url = self.create_connection_url(
+            database_name,
+            self.config.database_superusername,
+            self.config.database_superuserpassword
+        )
+
+        # normal user credentials
+        normal_user_pg_url = self.create_connection_url(
+            database_name,
+            self.config.database_username,
+            self.config.database_password
+        )
+
+        # ensure that if on Heroku the the normal_user_pg_url and the
+        # superuser_pg_url are the same
+        if self.config.on_heroku and (normal_user_pg_url != superuser_pg_url):
+            self.config.logger.error(
+                'there is no superuser (%s) when using Heroku',
+                self.config.database_superusername
+            )
+            return 1
+
+        # table logging section
+        if self.config.unlogged:
+            @compiles(CreateTable)
+            def create_table(element, compiler, **kw):
+                text = compiler.visit_create_table(element, **kw)
+                text = re.sub("^\sCREATE(.*TABLE)",
+                              lambda m: "CREATE UNLOGGED %s" %
+                              m.group(1), text)
+                return text
+
+        # Postgres version check section
+        self.config.logger.info('Postgres version check section with %s', superuser_pg_url)
+        with PostgreSQLAlchemyManager(
+            superuser_pg_url,
+            self.config.logger,
+            autocommit=False
+        ) as db:
+            if not db.min_ver_check(90200):
+                self.config.logger.error(
+                    'unrecognized PostgreSQL version: %s',
+                    db.version_string()
+                )
+                self.config.logger.error('Only 9.2+ is supported at this time')
+                return 1
+
+        # At this point, we're done with setup and we can perform actions requested by the user
+
+        # We can only do the following if the DB is not Heroku
+        # XXX Might add the heroku commands for resetting a DB here
+        if self.config.dropdb and not self.config.on_heroku:
+            # If this doesn't return True (aka everything worked fine), then return exit code 2
+            if not self.handle_dropdb(superuser_pg_url, database_name):
+                return 2
+
+        if self.config.createdb:
+            # If this doesn't return True (aka everything worked fine), then return exit code 1
+            if not self.handle_createdb(
+                superuser_normaldb_pg_url, superuser_pg_url, normal_user_pg_url, database_name
+            ):
+                return 1
```

Comment:
These 12 lines are different. Instead of doing all the work in `main`, they call out to `handle_` functions to do the work.